ruby/ql/lib/codeql/ruby/controlflow/internal/ControlFlowGraphImpl.qll

```diff
@@ -1414,9 +1416,6 @@ module Trees {
       last = this and
       c.isValidFor(this) and
       c.(ConditionalCompletion).getValue() = false
-      or
-      last(this.getBody(), last, c) and
-      c instanceof NormalCompletion
```

Comment:
Sorry; I was not clear: What I meant was that the `c instanceof NormalCompletion` should be removed; we still need `last(this.getBody(), last, c)` (I wonder why the CFG tests don't fail).

Reply:
Ah yes, it works because of the (somewhat non-compositional)
```rb
last(this.getABranch().(WhenClause).getBody(), pred, c)
```
I still think we should add this back, though.